addons/survey/static/src/js/survey_form.js

```diff
@@ -0,0 +1,272 @@
+        }
+    },
+
+    /*
+    * Convert the client side date format into server side date format before submit
+    */
+    _updateDateForSubmit: function (date_group, params) {
+        var input = $(date_group).find('input');
+        var date_value = input.val();
+        var questiontype = $(input).closest('.o_survey_form_date').data('questiontype');
+        if (date_value) {
+            var moment_date = questiontype === 'datetime' ? field_utils.parse.datetime(date_value, null, {timezone: true}) : field_utils.parse.date(date_value);
+            params[input.attr('name')] = moment_date ? moment_date.toJSON() : '';
+            input.val(params[input.attr('name')]);
+        }
+    },
+
+    /*
+    * Checks, for review mode, if the 'other' choice is checked.
+    * Clear the comment input if not checked (just in case) or focus on it.
+    */
+    _updateSimpleRadioForDisplay: function ($radio_group) {
+        var $other = $radio_group.find('.o_survey_form_radio_other');
+        var $comment_input = $radio_group.find('input[type="text"]');
+        if ($other.prop('checked')) {
+            $comment_input.focus();
+        } else {
+            $comment_input.val('');
+        }
+    },
+
+    /*
+    * Checks, for review mode, if the 'other' choice is checked.
+    * Clear and hide the comment input if not checked (just in case) or display it.
+    */
+    _updateSimpleSelectForDisplay: function ($select_group) {
+        var $select = $select_group.find('select');
+        var $other = $select_group.find('.o_survey_form_select_other');
+        var $comment_input = $select_group.find('input[type="text"]');
+        if ($select.val() === $other.val()) {
+            $comment_input.show();
+        } else {
+            $comment_input.val('');
+            $comment_input.hide();
+        }
+    },
+
+    // TOOLS
+    // -------------------------------------------------------------------------
+    /**
+    * Convert date value in client current timezone (if review mode)
+    */
+    _formatDateValue: function ($date_group) {
+        var input = $date_group.find('input');
+        var date_value = input.val();
+        if (date_value != '') {
+            var moment_date = field_utils.parse.date(date_value);
+            if ($date_group.data('questiontype') === 'datetime') {
+                date_value = field_utils.format.datetime(moment_date, null, {timezone: true});
+            } else {
+                date_value = field_utils.format.date(moment_date, null, {timezone: true});
+            }
+        }
+        input.val(date_value);
+        return date_value
+    },
+
+    /**
+    * Initialize datetimepicker in correct format and with constraints
+    */
+    _initDateTimePicker: function ($date_group) {
+        var disabledDates = []
+
+        var minDate = $date_group.data('mindate') ? field_utils.format.datetime(moment($date_group.data('mindate')), null, {timezone: true}) : moment({ y: 1900 });
```

Comment:
Maybe extract `field_utils.format.datetime(moment($date_group.data('mindate')), null, {timezone: true})`
to a `_formatDateTime` method that takes the target data as argument to avoid having really long lines.